.github/workflows/software-lint.yml

```
@@ -23,22 +18,21 @@ jobs:
         with:
           args: "--recursive ./src/software/firmware/"
 
-#   cppcheck:
-#     runs-on: ubuntu-18.04
-#     steps:
-#     - name: install cppcheck
-#       run: |
-#         sed 's/bionic/focal/g' /etc/apt/sources.list | sudo tee /etc/apt/sources.list.d/focal.list
-#         sudo apt-get update
-#         echo "Package: cppcheck" | sudo tee /etc/apt/preferences.d/cppcheck
-#         echo "Pin: release n=focal" | sudo tee -a /etc/apt/preferences.d/cppcheck
-#         echo "Pin-Priority: 990" | sudo tee -a /etc/apt/preferences.d/cppcheck
-#         sudo apt-get install -y cppcheck
-#     - uses: actions/checkout@v2
-#     - name: run cppcheck
-#       run: |
-#         cppcheck --version
-#         cppcheck --error-exitcode=1 --enable=all ./src/software/firmware/
-#         echo $?
-#         cppcheck --error-exitcode=1 --addon=cert ./src/software/firmware/
-#         #cppcheck --error-exitcode=1 --addon=misra ./src/software/firmware/
+  cppcheck:
+    runs-on: ubuntu-18.04
+    steps:
+    - name: install cppcheck
+      run: |
+        sed 's/bionic/focal/g' /etc/apt/sources.list | sudo tee /etc/apt/sources.list.d/focal.list
+        sudo apt-get update
+        echo "Package: cppcheck" | sudo tee /etc/apt/preferences.d/cppcheck
+        echo "Pin: release n=focal" | sudo tee -a /etc/apt/preferences.d/cppcheck
+        echo "Pin-Priority: 990" | sudo tee -a /etc/apt/preferences.d/cppcheck
+        sudo apt-get install -y cppcheck
+    - uses: actions/checkout@v2
+    - name: run cppcheck
```

what about splitting into 3 rules, something like that:
 ```suggestion
    - name: run cppcheck
      run: |
        cppcheck --version
        cppcheck --error-exitcode=1 --enable=all ./src/software/firmware/
    - name: run cppcheck cert
      if: success() || failure()
      run: |
        cppcheck --version
        cppcheck --error-exitcode=1 --addon=cert ./src/software/firmware/
    - name: run cppcheck misra
      if: success() || failure()
      run: |
        cppcheck --version
        cppcheck --error-exitcode=1 --addon=misra ./src/software/firmware/
```

---

sorry, probably not really useful if we dont user an error exitcode.

---

I'm not sure to get the point... I'll stay with my version except if you explain me what I missed here 😴

---

yeap I was not clear indeed. I thought that the option error-exitcode=1 was still on. So having a dedicated run step for each cppcheck call allowed to see which one had success/failed separately in the UI. Separating the calls also allow to check which log corresponds to which call more easily.
Not sure I am clearer :D